tests/integration/test_active_response/test_analysisd/test_os_exec.py

```diff
@@ -0,0 +1,351 @@
+    assert json_alert['parameters']['alert']['data']['dstuser'] == DST_USR, 'Invalid destination user in JSON message'
+    if extra_args == 'yes':
+        assert json_alert['parameters']['extra_args'], 'Missing extra_args in JSON message'
+        assert json_alert['parameters']['extra_args'][0] == ARG1, 'Missing arg1 in JSON message'
+        assert json_alert['parameters']['extra_args'][1] == ARG2, 'Missing arg1 in JSON message'
+
+
+# TESTS
+def test_os_exec(set_debug_mode, get_configuration, configure_environment, restart_service, configure_agents):
+    metadata = get_configuration.get('metadata')
+    protocol = metadata['protocol']
+    extra_args = metadata['extra_args']
+    timeout = metadata['timeout']
+    all_agents = metadata['all_agents']
+    sender = Sender(SERVER_ADDRESS, protocol=protocol)
+    log_monitor = FileMonitor(LOG_FILE_PATH)
+    injectors = []
+
+    # Agents
+    for agent in agents:
+        injector = Injector(sender, agent)
+        injectors.append(injector)
+        injector.run()
+        if protocol == "tcp":
+            sender = Sender(manager_address=SERVER_ADDRESS, protocol=protocol)
+
+    agents_id = [x.id for x in agents]
+
+    # Give time for registration key to be available and send a few heartbeats
+    time.sleep(30)
+
+    if all_agents == 'yes':
+        message = "1:(" + SERVER_NAME + ") " + LOG_LOCATION + ":" + LOG_MESSAGE
+        send_message(message, ANALYSISD_SOCKET)
+
+        for agent in agents:
+            # Checking AR in logs
+            try:
+                log_monitor.start(timeout=60, callback=wait_ar_line)
+            except TimeoutError as err:
+                raise AssertionError("AR message tooks too much!")
+
+            last_log = log_monitor.result()
+
+            if agent.os == 'ubuntu20.04':
+                # Version 4.2
+                validate_new_ar_message(agents_id, last_log, extra_args, timeout, all_agents)
+            else:
+                # Version < 4.2
+                validate_old_ar_message(agents_id, last_log, extra_args, timeout, all_agents)
+
+    else:
+        for agent in agents:
+            message = "1:[" + str(agent.id) + "] (" + agent.name + ") " + LOG_LOCATION + ":" + LOG_MESSAGE
+            send_message(message, ANALYSISD_SOCKET)
+
+            # Checking AR in logs
+            try:
+                log_monitor.start(timeout=60, callback=wait_ar_line)
+            except TimeoutError as err:
+                raise AssertionError("AR message tooks too much!")
+
+            last_log = log_monitor.result()
+
+            if agent.os == 'ubuntu20.04':
+                # Version 4.2
+                validate_new_ar_message(agent.id, last_log, extra_args, timeout, all_agents)
+            else:
+                # Version < 4.2
+                validate_old_ar_message(agent.id, last_log, extra_args, timeout, all_agents)
+
+    for injector in injectors:
+        injector.stop_receive()
```

Comment:
Documentation of the methods and the test is missing.

For example, take a look at the documentation of the vuln detector [module](https://github.com/wazuh/wazuh-qa/blob/master/deps/wazuh_testing/wazuh_testing/vulnerability_detector.py#L445) or any [related test](https://github.com/wazuh/wazuh-qa/blob/master/tests/integration/test_vulnerability_detector/test_scan_results/test_ubuntu_inventory_canonical_feed.py).